plugins/modules/virt.py

```diff
@@ -269,7 +271,10 @@ def destroy(self, vmid):
         return self.find_vm(vmid).destroy()
 
     def undefine(self, vmid, flag):
-        return self.find_vm(vmid).undefineFlags(flag)
+        vm = self.find_vm(vmid)
+        if flag & 32:
+            self.delete_domain_volumes(vmid)
+        return vm.undefineFlags(flag)
```

Comment:
```suggestion
        vm = self.find_vm(vmid)
        if flag & 32:
            self.delete_domain_volumes(vmid)
        return vm.undefineFlags(flag)
```
I'm not a developer of this collection, but i wonder why this code is here and not in a dedicated method like you can see a bit above?

Reply:
I'm not sure I fully understand this question. However, if the flag check is made it's own method, which it could very well be, these 2 lines would move into another function and then would be called inside the `undefine()` method anyways as far as I can see. 

I suppose the other option is to put the check in `core()` under the `undefine` if statement. It would expand the `if` statement in [line 780](https://github.com/ansible-collections/community.libvirt/blob/126062514e36351f71fe50cf621c6ba5c900e148/plugins/modules/virt.py#L780)

Reply:
I'll mark it as resolved, no action is needed, thanks